File: ios/NativeModulesWorkshops/ChartsModule/ChartsModule.m
//
//  ChartsModule.m
//  NativeModulesWorkshops
//
//  Created by Agata Kosior on 23/09/2024.
//

#import <React/RCTBridgeModule.h>
#import <React/RCTBridge.h>

@interface RCT_EXTERN_MODULE(ChartsModule, NSObject)

RCT_EXTERN_METHOD(isRotationEnabled:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(enableRotation:(nonnull NSNumber *)enable)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/NativeModulesWorkshops/ChartsModule/ChartsModule.swift
//
//  ChartsModule.swift
//  NativeModulesWorkshops
//
//  Created by Agata Kosior on 23/09/2024.
//

import Foundation

@objc(ChartsModule)
class ChartsModule: NSObject {
  var enabled = true;

  @objc(isRotationEnabled:withRejecter:)
  func isRotationEnabled(resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(enabled)
  }

  @objc(enableRotation:)
  func enableRotation(enable: NSNumber) -> Void {
    enabled = enable.boolValue
    NotificationCenter.default.post(name: Notification.Name(enableRotationNotification), object: enabled)
  }
}
